src/compiler/config_protobuf.h

```diff
@@ -49,4 +49,14 @@
   ::google::protobuf::compiler::ParseGeneratorParameter
 #endif
 
+#ifndef GRPC_CUSTOM_CSHARP_GETCLASSNAME
+#include <google/protobuf/compiler/csharp/csharp_names.h>
+#define GRPC_CUSTOM_CSHARP_GETCLASSNAME \
+  ::google3_proto_compat::compiler::csharp::GetClassName
```

Comment:
I think in the OSS version of this file, you need to use the `::google::protobuf::compiler::csharp::` namespace,
so that the csharp_generator.cc stays "unchanged" after replacing the `GRPC_CUSTOM_CSHARP_*`  macros.

Reply:
Updated